### DISCUSSION TITLE
Components load visible, and then disappear once Alpine loads

### DISCUSSION THREAD

**Discussion by LloydThinks** [alpinejs/alpine#777] [4. General]
Hi! I am new to AlpineJS, and I something worries me with what little expose I have so far. Every component with `x-show="open"` and some parent component with `x-data="{ open: false }"` loads VISIBLE at first, and then AlpineJS appears to catch up, and then they all disappear.

Is there a way to load the page with them not visible? After some time searching for "pre-loading" or some kind of "load page AFTER Alpine data loads" have returned no results.

Thank you for making such a beautiful JS interface, and I hope to resolve this one little thing.
Lloyd

**Comment by ryangjchandler** [comment]
@LloydThinks Add `x-cloak` to your elements that have `x-show` on them and add the following bit of CSS to your site too:

```css
[x-cloak] {
    display: none !important;
}
```

**Reply by LloydThinks** [reply]
Thank you very much! This was exactly what I was looking for.

Cheers!

**Reply by caner-aras** [reply]
thanks a lot! saved me.

**Reply by rajivmehta-github** [reply]
This was a pain area, a race condition, which is rectified with this workaround

**Reply by ekwoka** [reply]
This isn't a workaround. It's the official way :)
